jwst/cube_build/data_types.py

```diff
@@ -1,119 +1,107 @@
-""" Class Data Type is used to read in the input data. It also determines
-if the input data is a single science exposure, an association table, a
-single datamodel or several data models stored in a ModelContainer.
-"""
-import os
+# Class Data Type is used to read in the input data.
 
-from stdatamodels.jwst import datamodels
+# It also determines if the input data is a single science exposure, an association table, a
+# single datamodel or several data models stored in a ModelContainer.
 
+from pathlib import PurePath
+from stdatamodels.jwst import datamodels
 from jwst.datamodels import ModelContainer
-
 import logging
+
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
 
 # ******************************************************************************
-class DataTypes():
-
-    """ Class to handle reading input data to cube_build.
-    """
-
-    template = {"asn_rule": "",
-                "target": "",
-                "asn_pool": "",
-                "asn_type": "",
-                "products": [
-                    {"name": "",
-                     "members": [
-                         {"exptype": "",
-                          "expname": ""}
-                     ]
-                     }
-                ]
-                }
-
-    def __init__(self, input, single, output_file, output_dir):
-        """ Read in input data and determine what type of input data.
+class DataTypes:
+    """Class to handle reading input data to cube_build."""
+
+    template = {
+        "asn_rule": "",
+        "target": "",
+        "asn_pool": "",
+        "asn_type": "",
+        "products": [{"name": "", "members": [{"exptype": "", "expname": ""}]}],
+    }
+
+    def __init__(self, input_data, single, output_file, output_dir):
+        """
+        Read in input data and determine what type of input data.
```

Comment:
This is a little unclear - maybe revise the summary as:
```suggestion
        Assemble input data for processing.
```
and then add in the text from the top of the module.